ts/features/bonus/cgn/components/LegacyCgnServiceCTA.tsx

```diff
@@ -0,0 +1,111 @@
+import * as React from "react";
+import { useEffect, useRef } from "react";
+import * as pot from "@pagopa/ts-commons/lib/pot";
+import { Alert } from "react-native";
+import { constNull } from "fp-ts/lib/function";
+import { Label } from "../../../../components/core/typography/Label";
+import ButtonDefaultOpacity from "../../../../components/ButtonDefaultOpacity";
+import I18n from "../../../../i18n";
+import { useIODispatch, useIOSelector } from "../../../../store/hooks";
+import { servicePreferenceSelector } from "../../../services/store/reducers/servicePreference";
+import { isServicePreferenceResponseSuccess } from "../../../services/types/ServicePreferenceResponse";
+import { ServiceId } from "../../../../../definitions/backend/ServiceId";
+import { cgnActivationStart } from "../store/actions/activation";
+import { cgnUnsubscribe } from "../store/actions/unsubscribe";
+import { fold, isLoading } from "../../../../common/model/RemoteValue";
+import { showToast } from "../../../../utils/showToast";
+import { cgnUnsubscribeSelector } from "../store/reducers/unsubscribe";
+import { loadServicePreference } from "../../../services/store/actions";
+import ActivityIndicator from "../../../../components/ui/ActivityIndicator";
+import { loadAvailableBonuses } from "../../common/store/actions/availableBonusesTypes";
+
+type Props = {
+  serviceId: ServiceId;
+};
+const LegacyCgnServiceCTA = (props: Props) => {
+  const isFirstRender = useRef<boolean>(true);
+  const dispatch = useIODispatch();
+  const servicePreference = useIOSelector(servicePreferenceSelector);
+  const unsubscriptionStatus = useIOSelector(cgnUnsubscribeSelector);
+
+  const servicePreferenceValue = pot.getOrElse(servicePreference, undefined);
+
+  useEffect(() => {
+    if (!isFirstRender.current) {
+      fold(
+        unsubscriptionStatus,
+        constNull,
+        constNull,
+        () => {
+          showToast(I18n.t("bonus.cgn.activation.deactivate.toast"), "success");
+          dispatch(loadServicePreference.request(props.serviceId));
+        },
+        () => {
+          showToast(I18n.t("global.genericError"), "danger");
+        }
+      );
+    }
+    // eslint-disable-next-line functional/immutable-data
+    isFirstRender.current = false;
+  }, [unsubscriptionStatus, dispatch, props.serviceId]);
+
+  if (
```

Comment:
as above

Reply:
All components with the `Legacy` prefix will be removed, so I did not refactor them for this reason